src/pybind/mgr/dashboard/frontend/src/app/shared/components/notification-banner/notification-banner.component.ts

```diff
@@ -0,0 +1,148 @@
+      this.telemetryNotificationService.setVisibility(false);
+    } else if (this.componentType === 'motd') {
+      this.motdSubscription.unsubscribe();
+    }
+  }
+
+  onDismissed(): void {
+    if (this.componentType === 'pwd-expiration') {
+      this.authStorageService.isPwdDisplayedSource.next(false);
+    } else if (this.componentType === 'telemetry') {
+      this.telemetryNotificationService.setVisibility(false);
+      localStorage.setItem('telemetry_notification_hidden', 'true');
+      this.notificationService.show(
+        NotificationType.success,
+        $localize`Telemetry activation reminder muted`,
+        $localize`You can activate the module on the Telemetry configuration \
+        page (<b>Dashboard Settings</b> -> <b>Telemetry configuration</b>) at any time.`
+      );
+    } else if (this.componentType === 'motd') {
+      this.motdNotificationService.hide();
+    }
+    this.displayNotification = false;
+  }
+
+  private getExpirationDays(pwdExpirationDate: number): number {
+    const current = new Date();
+    const expiration = new Date(pwdExpirationDate * 1000);
+    return Math.floor((expiration.valueOf() - current.valueOf()) / (1000 * 3600 * 24));
+  }
+
+  isNotificationHidden(): boolean {
+    return localStorage.getItem('telemetry_notification_hidden') === 'true';
+  }
+
+  setPwdExpirationSubscription(): void {
+    this.settingsService.getStandardSettings().subscribe((pwdExpirationSettings) => {
+      this.pwdExpirationSettings = new CdPwdExpirationSettings(pwdExpirationSettings);
+      const pwdExpirationDate = this.authStorageService.getPwdExpirationDate();
+      if (pwdExpirationDate) {
+        this.expirationDays = this.getExpirationDays(pwdExpirationDate);
+        if (this.expirationDays <= this.pwdExpirationSettings.pwdExpirationWarning2) {
+          this.notificationType = 'danger';
+        } else {
+          this.notificationType = 'warning';
+        }
+        this.displayNotification =
+          this.expirationDays <= this.pwdExpirationSettings.pwdExpirationWarning1;
+        this.authStorageService.isPwdDisplayedSource.next(this.displayNotification);
+
+        if (this.expirationDays === 0) {
+          this.notificationMessage =
+            '<div i18n>Your password will expire in <strong>less than 1</strong> day. Click <a href="#/user-profile/edit" class="alert-link">here</a> to change it now.</div>';
+        } else if (this.expirationDays > 0) {
+          this.notificationMessage =
+            '<div i18n>Your password will expire in <strong>' +
+            this.expirationDays +
+            '</strong> day(s). Click <a href="#/user-profile/edit" class="alert-link">here</a> to change it now.</div>';
+        }
+      }
+    });
+  }
+
+  setMotdSubscription(): void {
+    this.motdSubscription = this.motdNotificationService.motd$.subscribe(
+      (motd: Motd | undefined) => {
+        if (motd) {
+          this.notificationType = motd.severity;
```

Comment:
This line clearly points to the fact that `notificationType` property should be renamed to `notificationLevel` or `notificationSeverity`.